src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -1,38 +1,12 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
-/**
- * Parameterize this service and add its implementation in a separate class.
- */
-public interface MachineService {
+public interface MachineService<T extends Machine> {
+    List<Machine> getAll(Class<? extends Machine> type);
 
-    /**
-     * Return the list of machines.
-     * In the implementation of this method please use your MachineProducer implementations
-     * See that 'Class type' is not parametrized.
-     * Consider to parametrize this Class< PARAMETRIZE_ME > with specific wildcard bounding
-     *
-     * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
-     * @return the list of machines
-     */
-    List<Object> getAll(Class type);
+    void fill(List<? super Machine> machines, T value);
```

Comment:
In the `fill` method, the `machines` list parameter should be parameterized with `T` instead of `Machine` to ensure that the list can only contain instances of `T` or its subtypes. This will maintain type safety according to the generic class definition.